src/compiler/diagnosticMessages.json

```diff
@@ -3960,5 +3960,17 @@
     "Convert to ES6 module": {
         "category": "Message",
         "code": 95017
+    },
+    "add Undefined type to property '{0}'": {

```

Comment:
Nit: Start with a capital letter, but don't capitalize other things (e.g. say "definite assignment assertion" not "Definite Assignment Assertion".) Also put `'undefined'` in single quotes.